workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Injection/IInjectableCreator.cs

```diff
@@ -0,0 +1,12 @@
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Interface for objects creating a particular type of IInjectable, to be used by the InjectableFactory.
+    /// </summary>
+    internal interface IReaderWriterCreator
```

Comment:
I think it's this PR that should continue the work within `IInjectableCreator.cs` a tiny bit more and rename `IReaderWriterCreator` and accept a `InjectableType` parameter, implementing the `ReaderWriter` part and `NotImplementedException` with `// TODO {ticketid}` for the rest.

Reply:
Yes I missed some renames there, will change. The intention was to have separate Creators for each specific type though, ie one for the ReaderWriter, one for each command's sender etc. To have one per component we'd have to add some additional parameters beside the InjectableType too to specify the command index, but having separate Creators for each is straightforward to do.

Reply:
Could you pass in the `InjectableId` as a parameter to the `CreateInjector` function and then the creator method could be as sophisticated as necessary

Reply:
Could something like this work for example

```
            public IInjectable CreateInjectable(Entity entity, InjectableId injectableId, EntityManager entityManager, ILogDispatcher logDispatcher)
            {
                switch(injectableId.type) {
                    case Reader: return new ReaderWriterImpl(entity, entityManager, logDispatcher);
                    default: NotImplementedYet();
                }
           }
```

Reply:
Yes, and then Creators would have an attribute to associate them with a component ID rather than an injectable ID still, that could work. Is there a good reason to do that though? It feels cleaner to have just one for each and not pass into them info (part of which was actually used for identifying the appropriate creator in the first place..)

Reply:
It depends whether we'd like to increase the complexity in the factory vs next to the accessors. I am unsure how cleaner it would be to do the former, can you show what the code could look like that way?

Reply:
The factory literally wouldn't have to change at all, since it's just mapping from InjectableId to Creator, and the InjectableIds identify the specific type uniquely (including different command senders on same component etc). 

So the Factory code would look as it does now, and we'd have for each component a ReaderWriterCreator and for each command 3 separate Creators (Sender, RequestHandler, ResponseHandler), all with oneliner implementations like the current ReaderWriterCreator.

Reply:
Doesn't sound too bad 👍 can you create stubs in this PR?

Reply:
Good idea